src/odemis/driver/tfsbc.py

```diff
@@ -0,0 +1,130 @@
+        :param rs485_serial_num (str): serial number of RS485 adapter
+        """
+        # .hwVersion, .swVersion not available
+        model.HwComponent.__init__(self, name, role, **kwargs)
+
+        self._amps_per_m = amps_per_m
+        self._zero_offset_amps = zero_offset_amps
+
+        # Find port by RS485 adapter serial number
+        for p in serial.tools.list_ports.comports():
+            if rs485_serial_num in p.hwid:
+                # Found it!
+                port = p
+                break
+        else:
+            raise IOError("RS484 adapter with serial number %s not found!" % rs485_serial_num)
+
+        # Modbus connection
+        self._modbus = ModbusSerialClient(method='rtu', port=port,
+                                          baudrate=BAUDRATE, timeout=TIMEOUT,
+                                          stopbits=STOPBITS, parity=PARITY,
+                                          bytesize=BYTESIZE)
+
+        # Calculate range
+        range_amps = numpy.subtract(RANGE_AMPS, zero_offset_amps)
+        range_m = numpy.divide(range_amps, amps_per_m)
+
+        # Default value: zero offset (in m)
+        default_shift = tuple(numpy.divide(zero_offset_amps, amps_per_m))
+        self.beamShift = model.TupleContinuous(default_shift, range_m,
+                                           cls=(int, float), unit="m",
+                                           setter=self._setShift)
+
+    def _setShift(self, value):
+        """
+        :param value (float, float): x, y shift from the center (in m)
+        """
+        # Convert meter to amps
+        value = numpy.multiply(value, self._amps_per_m)
+
+        # Subtract offset (center amplitude)
+        value = numpy.subtract(value, self._zero_offset_amps)
+
+        # Convert amps to 16-bit int
+        value = numpy.subtract(value, RANGE_AMPS[0])
+        value = int(numpy.multiply(value, 0xFFFF / (RANGE_AMPS[1] - RANGE_AMPS[0])))
+        # Note: in the example script a constant of 0.5 is added before converting
+        # to int to always round up (why?)
+
+        payload = [-value[0], -value[1], value[0], value[1]]
+
+        self._write_registers(payload)
+
+    def _write_registers(self, payload):
+        """
+        Write to all four registers. Try to reconnect to device in case connection was lost.
+        :payload (list of 4 ints): register values (-x, -y, x, y)
+        """
+        if len(payload) != 4:
+            raise ValueError("write_registers received payload of invalid length %s != 4." % len(payload))
+        try:
+            # write all registers together (starting at lower x register (=0x01))
+            self._modbus.write_registers(BEAMDEFL_LX, payload, unit=SLAVE_UNIT)
+        except IOError:
+            logging.warning("Failed to write registers, trying to reconnect...")
+            self._modbus.close()
+            self._modbus.connect()
+            logging.info("Recovered device.")
```

Comment:
If you want to handle reconnection, let's do it completely. Please move this to a separate function, and inside:
* Update the .state to indicate there is a connection error
* Try reconnecting *in a loop*
* Put back the state when it's reconnected.